index.js

```diff
@@ -1,105 +1,76 @@
 'use strict';
 
-var superagent = require('superagent'),
+var getJwt = require('frau-jwt'),
+	superagent = require('superagent'),
+	url = require('url'),
 	xsrf = require('frau-superagent-xsrf-token');
 
-var accessTokenExpiresAt = 0,
-	oauth2Enabled = true;
+function noop () {}
 
-function now() {
-	return Date.now()/1000 | 0;
+function isRelative/*ly safe*/ (url) {
+	return url.hostname === null;
 }
 
-function addHeaders(req) {
-	req = req.use(xsrf);
-	req.set('X-D2L-App-Id', 'deprecated');
-	return req;
+function endsWith (haystack, needle) {
+	var expectedPosition = haystack.length - needle.length;
+	var lastIndex = haystack.indexOf(needle, expectedPosition);
+	var result = lastIndex !== -1 && lastIndex === expectedPosition;
+	return result;
 }
 
-function processRefreshResponse(err, res) {
-	// Prior to OAuth 2 support the refreshcookie route returns 404.
-	if(res.status == 404) {
-		disableOAuth2();
-		return;
-	}
-
-	// In the future we should log an error
-	if(err || !res.ok) {
-		return;
-	}
-
-	var cacheControl = res.headers['cache-control'];
-	if(!cacheControl) {
-		return;
-	}
+function isBrightspaceApi (url) {
+	return url.protocol === 'https:'
+		&& (url.hostname === 'api.brightspace.com'
```

Comment:
Ok, well your call on whether we need that check.

It also occurs to me that it gets weird when we are developing services. We could always have a policy of reverse proxying our dev machines via *.api.brightspace.com, or allow sending tokens to localhost, but yeah. That can be a separate discussion.